xbmc/music/Artist.h

```diff
@@ -130,21 +132,49 @@ class CArtistCredit
 
   std::string GetArtist() const                { return m_strArtist; }
   std::string GetMusicBrainzArtistID() const   { return m_strMusicBrainzArtistID; }
-  std::string GetJoinPhrase() const            { return m_strJoinPhrase; }
   int         GetArtistId() const              { return idArtist; }
   void SetArtist(const std::string &strArtist) { m_strArtist = strArtist; }
   void SetMusicBrainzArtistID(const std::string &strMusicBrainzArtistID) { m_strMusicBrainzArtistID = strMusicBrainzArtistID; }
-  void SetJoinPhrase(const std::string &strJoinPhrase) { m_strJoinPhrase = strJoinPhrase; }
   void SetArtistId(int idArtist)               { this->idArtist = idArtist; }
 
 private:
   long idArtist;
   std::string m_strArtist;
   std::string m_strMusicBrainzArtistID;
-  std::string m_strJoinPhrase;
-  bool m_boolFeatured;
 };
 
 typedef std::vector<CArtist> VECARTISTS;
 typedef std::vector<CArtistCredit> VECARTISTCREDITS;
 
+#define ROLE_ARTIST 1  //Default role
+
+class CMusicRole
+{
+public:
+  CMusicRole() { }
+  CMusicRole(std::string strRole, std::string strArtist) : idRole(-1), m_strRole(strRole), m_strArtist(strArtist), idArtist(-1) { }
+  CMusicRole(int role, std::string strRole, std::string strArtist, long ArtistId) : idRole(role), m_strRole(strRole), m_strArtist(strArtist), idArtist(ArtistId) { }
+  std::string GetArtist() const { return m_strArtist; }
+  std::string GetRoleDesc() const { return m_strRole; }
+  int GetRoleId() const { return idRole; }
+  long GetArtistId() const { return idArtist; }
+  void SetArtistId(const long iArtistId) { idArtist = iArtistId;  }
+
+  bool operator==(const CMusicRole& a) const
+  {
+    if (StringUtils::EqualsNoCase(m_strRole, a.m_strRole))
+      return StringUtils::EqualsNoCase(m_strArtist, a.m_strArtist);
+    else
+      return false;
+  }
+private:
+  int idRole;
+  std::string m_strRole;
+  std::string m_strArtist;
+  long idArtist;
+};
+
+typedef std::vector<CMusicRole> VECMUSICROLES;
```

Comment:
not really needed, I personally don't like these typedefs but some disagree so you can decide yourself what's best :)